Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift

```diff
@@ -125,6 +132,19 @@ public struct AutomaticCuration {
         
         let parentReference = canonicalPath.last!
         
+        func filterReferences(_ references: [ResolvedTopicReference]) throws -> [ResolvedTopicReference] {
+            try references
+                // Don't include the current node.
+                .filter { $0 != node.reference }
+            
+                // Filter out nodes that aren't available in the given trait.
+                .filter { reference in
+                    try context.entity(with: reference)
```

Comment:
Should we adopting this logic in the `topics(for:withTrait:context:)` function above? It seems cleaner than only relying on the kind being available for a given trait but maybe I'm missing something. (I'm not entirely sure what `availableVariantTraits` is pulled from.)

Reply:
We could, but we need to get the appropriate language-specific kind of the symbol anyway in that case. If we ported this logic to `topics(for:withTrait:context:)`, we'd check the languages in `DocumentationNode. availableVariantTraits` but also need to verify that the symbol has `kindVariants` for each of these languages which seems redundant. Compilers are expected to produce the `kind` field for their symbols, so DocC shouldn't be in a case where a symbol is available in multiple languages, but the symbol only has 1 `kind` value.

`DocumentationNode.availableVariantTraits` is populated from this logic: https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/Symbol%20Graph/SymbolReference.swift#L174

Reply:
> Compilers are expected to produce the kind field for their symbols, so DocC shouldn't be in a case where a symbol is available in multiple languages, but the symbol only has 1 kind value.

What if we have the same kind in multiple languages?